src/Databases/DatabasesCommon.cpp

```diff
@@ -258,6 +288,19 @@ StoragePtr DatabaseWithOwnTablesBase::detachTableUnlocked(const String & table_n
         throw Exception(ErrorCodes::UNKNOWN_TABLE, "Table {}.{} doesn't exist",
                         backQuote(database_name), backQuote(table_name));
     res = it->second;
```

Comment:
How about getting rid of `first`, `second` by means of
```
auto & [table_name, table_storage] = *it;
```

Reply:
I dont mind. However I suggest to remove res (becasue we can use table_storage) and use
```
auto & [_, table_storage] = *it;
```
because we already have table_name.